arches_lingo/views/api/concepts.py

```diff
@@ -91,3 +93,68 @@ def default_sensitivity():
         if elastic_prefix_length >= 5:
             return 0
         return int(5 - elastic_prefix_length)
+
+
+@method_decorator(
+    group_required("RDM Administrator", raise_exception=True), name="dispatch"
+)
+class ConceptResourceView(ConceptTreeView):
+    def get(self, request):
+        scheme = request.GET.get("scheme", None)
+        exclude = request.GET.get("exclude", None)
+        term = request.GET.get("term", None)
+        page_number = request.GET.get("page", 1)
+        items_per_page = request.GET.get("items", 25)
+        concepts = request.GET.get("concepts", None)
+        concept_ids = concepts.split(",") if concepts else None
+        Concept = SemanticResource.as_model("concept")
+
+        if concept_ids:
+            concept_ids = concept_ids
+        else:
+            if scheme:
+                if exclude == "true":
+                    concept_query = Concept.exclude(
+                        part_of_scheme__0__0__resourceId=scheme
+                    )
+                else:
+                    concept_query = Concept.filter(
+                        part_of_scheme__0__0__resourceId=scheme
+                    )
+                concept_ids = (
+                    concept_query.order_by("pk").values_list("pk", flat=True).distinct()
+                )
+            else:
+                concept_ids = (
+                    VwLabelValue.objects.all()
+                    .order_by("concept_id")
+                    .values_list("concept_id", flat=True)
+                    .distinct()
+                )
+
+        if term:
+            filtering_concept_ids = VwLabelValue.objects.filter(
+                value__icontains=term
+            ).values_list("concept_id", flat=True)
+            concept_ids = [id for id in concept_ids if id in filtering_concept_ids]
```

Comment:
To avoid eagerly evaluating the `concept_ids` query, which could end up materializing more rows than the items per page limit:

```suggestion
            concept_ids = concept_ids.filter(concept_id__in=filtering_concept_ids)
```

But to do this I think you'd need to standardize on what model `concept_ids` is querying--Concept/SemanticResource or VwLabelValue?--so you can keep chaining things on to it. (I know this is made more difficult since label searching predated the arches-querysets stuff.) Good news: I think you may not even need the concept_ids query on L128 since it's not filtering anything out anyway.